src/gui_common/CustomCheckBox.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Godot;
+
+/// <summary>
+///   A customized check box that changes icon when hovered / clicked.
+/// </summary>
+public class CustomCheckBox : ToolButton
+{
+    private Texture unpressedNormal;
+    private Texture unpressedHover;
+    private Texture unpressedClick;
+    private Texture pressedNormal;
+    private Texture pressedHover;
+    private Texture pressedClick;
+
+    private bool pressing;
+
+    public override void _Ready()
+    {
+        unpressedNormal = GD.Load<Texture>("res://assets/textures/gui/bevel/checkA.png");
+        unpressedHover = GD.Load<Texture>("res://assets/textures/gui/bevel/checkAhover.png");
+        unpressedClick = GD.Load<Texture>("res://assets/textures/gui/bevel/checkAclick.png");
+        pressedNormal = GD.Load<Texture>("res://assets/textures/gui/bevel/checkB.png");
+        pressedHover = GD.Load<Texture>("res://assets/textures/gui/bevel/checkBhover.png");
+        pressedClick = GD.Load<Texture>("res://assets/textures/gui/bevel/checkBclick.png");
```

Comment:
There's a better way to do this. Add a custom item to thrive theme file for CheckButton and use `GetIcon` to get that icon when updating this button's display icon.